### PR TITLE
Upgrade rust to 1.27.0

### PR DESCRIPTION
**Pull request by sfackler**

### CODE REVIEW

**Comment by yosifkit** [comment]
<details>
<summary>Diff:</summary>

```diff
diff --git a/_bashbrew-arches b/_bashbrew-arches
index 8ec1e8b..e00641c 100644
--- a/_bashbrew-arches
+++ b/_bashbrew-arches
@@ -1,16 +1,16 @@
-rust:jessie @ amd64
-rust:jessie @ arm32v7
-rust:jessie @ arm64v8
-rust:jessie @ i386
-rust:latest @ amd64
-rust:latest @ arm32v7
-rust:latest @ arm64v8
-rust:latest @ i386
-rust:slim @ amd64
-rust:slim @ arm32v7
-rust:slim @ arm64v8
-rust:slim @ i386
-rust:slim-jessie @ amd64
-rust:slim-jessie @ arm32v7
-rust:slim-jessie @ arm64v8
-rust:slim-jessie @ i386
+rust:1.27.0 @ amd64
+rust:1.27.0 @ arm32v7
+rust:1.27.0 @ arm64v8
+rust:1.27.0 @ i386
+rust:1.27.0-jessie @ amd64
+rust:1.27.0-jessie @ arm32v7
+rust:1.27.0-jessie @ arm64v8
+rust:1.27.0-jessie @ i386
+rust:1.27.0-slim @ amd64
+rust:1.27.0-slim @ arm32v7
+rust:1.27.0-slim @ arm64v8
+rust:1.27.0-slim @ i386
+rust:1.27.0-slim-jessie @ amd64
+rust:1.27.0-slim-jessie @ arm32v7
+rust:1.27.0-slim-jessie @ arm64v8
+rust:1.27.0-slim-jessie @ i386
diff --git a/_bashbrew-list b/_bashbrew-list
index 1000895..aa8e99a 100644
--- a/_bashbrew-list
+++ b/_bashbrew-list
@@ -1,24 +1,6 @@
-rust:1
-rust:1-jessie
-rust:1-slim
-rust:1-slim-jessie
-rust:1-slim-stretch
-rust:1-stretch
-rust:1.26
-rust:1.26-jessie
-rust:1.26-slim
-rust:1.26-slim-jessie
-rust:1.26-slim-stretch
-rust:1.26-stretch
-rust:1.26.2
-rust:1.26.2-jessie
-rust:1.26.2-slim
-rust:1.26.2-slim-jessie
-rust:1.26.2-slim-stretch
-rust:1.26.2-stretch
-rust:jessie
-rust:latest
-rust:slim
-rust:slim-jessie
-rust:slim-stretch
-rust:stretch
+rust:1.27.0
+rust:1.27.0-jessie
+rust:1.27.0-slim
+rust:1.27.0-slim-jessie
+rust:1.27.0-slim-stretch
+rust:1.27.0-stretch
diff --git a/rust_jessie/Dockerfile b/rust_1.27.0-jessie/Dockerfile
similarity index 98%
rename from rust_jessie/Dockerfile
rename to rust_1.27.0-jessie/Dockerfile
index f224faa..3e5d97c 100644
--- a/rust_jessie/Dockerfile
+++ b/rust_1.27.0-jessie/Dockerfile
@@ -3,7 +3,7 @@ FROM buildpack-deps:jessie
 ENV RUSTUP_HOME=/usr/local/rustup \
     CARGO_HOME=/usr/local/cargo \
     PATH=/usr/local/cargo/bin:$PATH \
-    RUST_VERSION=1.26.2
+    RUST_VERSION=1.27.0
 
 RUN set -eux; \
     \
diff --git a/rust_slim-jessie/Dockerfile b/rust_1.27.0-slim-jessie/Dockerfile
similarity index 98%
rename from rust_slim-jessie/Dockerfile
rename to rust_1.27.0-slim-jessie/Dockerfile
index 7e0e5dc..0c1c347 100644
--- a/rust_slim-jessie/Dockerfile
+++ b/rust_1.27.0-slim-jessie/Dockerfile
@@ -3,7 +3,7 @@ FROM debian:jessie-slim
 ENV RUSTUP_HOME=/usr/local/rustup \
     CARGO_HOME=/usr/local/cargo \
     PATH=/usr/local/cargo/bin:$PATH \
-    RUST_VERSION=1.26.2
+    RUST_VERSION=1.27.0
 
 RUN set -eux; \
     apt-get update; \
diff --git a/rust_slim/Dockerfile b/rust_1.27.0-slim/Dockerfile
similarity index 98%
rename from rust_slim/Dockerfile
rename to rust_1.27.0-slim/Dockerfile
index 1bd6725..390dbd9 100644
--- a/rust_slim/Dockerfile
+++ b/rust_1.27.0-slim/Dockerfile
@@ -3,7 +3,7 @@ FROM debian:stretch-slim
 ENV RUSTUP_HOME=/usr/local/rustup \
     CARGO_HOME=/usr/local/cargo \
     PATH=/usr/local/cargo/bin:$PATH \
-    RUST_VERSION=1.26.2
+    RUST_VERSION=1.27.0
 
 RUN set -eux; \
     apt-get update; \
diff --git a/rust_latest/Dockerfile b/rust_1.27.0/Dockerfile
similarity index 98%
rename from rust_latest/Dockerfile
rename to rust_1.27.0/Dockerfile
index e1b8299..cefddb8 100644
--- a/rust_latest/Dockerfile
+++ b/rust_1.27.0/Dockerfile
@@ -3,7 +3,7 @@ FROM buildpack-deps:stretch
 ENV RUSTUP_HOME=/usr/local/rustup \
     CARGO_HOME=/usr/local/cargo \
     PATH=/usr/local/cargo/bin:$PATH \
-    RUST_VERSION=1.26.2
+    RUST_VERSION=1.27.0
 
 RUN set -eux; \
     \
```

</details>

**Comment by yosifkit** [comment]
Build test of #4484; 13031010; `amd64` (`rust`):

```console

$ bashbrew build rust:1.27.0-stretch
Building bashbrew/cache:622f00a95a23b51f26801e09340c59330dfccd89f31eddcfdb2c880a7a093b05 (rust:1.27.0-stretch)
Tagging rust:1.27.0-stretch
Tagging rust:1.27.0

$ test/run.sh rust:1.27.0-stretch
testing rust:1.27.0-stretch
	'utc' [1/5]...passed
	'cve-2014--shellshock' [2/5]...passed
	'no-hard-coded-passwords' [3/5]...passed
	'override-cmd' [4/5]...passed
	'rust-hello-world' [5/5]...passed


$ bashbrew build rust:1.27.0-slim-stretch
Building bashbrew/cache:c4a876e142954294f3a8a40335ffd68819e1330dd6123a1dd177c641fbfd2228 (rust:1.27.0-slim-stretch)
Tagging rust:1.27.0-slim-stretch
Tagging rust:1.27.0-slim

$ test/run.sh rust:1.27.0-slim-stretch
testing rust:1.27.0-slim-stretch
	'utc' [1/5]...passed
	'cve-2014--shellshock' [2/5]...passed
	'no-hard-coded-passwords' [3/5]...passed
	'override-cmd' [4/5]...passed
	'rust-hello-world' [5/5]...passed


$ bashbrew build rust:1.27.0-jessie
Building bashbrew/cache:2426c4eab54017759c7b672df7baee7826f887b8a8762c3c0bb03a7da36acdfa (rust:1.27.0-jessie)
Tagging rust:1.27.0-jessie

$ test/run.sh rust:1.27.0-jessie
testing rust:1.27.0-jessie
	'utc' [1/5]...passed
	'cve-2014--shellshock' [2/5]...passed
	'no-hard-coded-passwords' [3/5]...passed
	'override-cmd' [4/5]...passed
	'rust-hello-world' [5/5]...passed


$ bashbrew build rust:1.27.0-slim-jessie
Building bashbrew/cache:4680d5a70da9611f59017c844a4f31c50987a14c54ee55a864c511694d824394 (rust:1.27.0-slim-jessie)
Tagging rust:1.27.0-slim-jessie

$ test/run.sh rust:1.27.0-slim-jessie
testing rust:1.27.0-slim-jessie
	'utc' [1/5]...passed
	'cve-2014--shellshock' [2/5]...passed
	'no-hard-coded-passwords' [3/5]...passed
	'override-cmd' [4/5]...passed
	'rust-hello-world' [5/5]...passed

```

File: library/rust [hunk 1]
```diff
@@ -3,22 +3,22 @@
 Maintainers: Steven Fackler <sfackler@gmail.com> (@sfackler)
 GitRepo: https://github.com/rust-lang-nursery/docker-rust.git
 
-Tags: 1.26.2-stretch, 1-stretch, 1.26-stretch, stretch, 1.26.2, 1, 1.26, latest
+Tags: 1.27.0-stretch, 1.27.0
```
Comment (yosifkit) [inline library/rust]: I assume the dropping and not adding generic tags like `1-stretch` and `1.27` was on purpose.

Did you also mean to drop `latest`? It is the tag pulled if a user just does `docker pull rust`. As is, they will continue to get `1.26.2-stretch` if they forget to specify a tag.

Comment (sfackler) [reply library/rust]: Gah, nope that's my bad. Will fix.

Comment (sfackler) [reply library/rust]: Done!

**Comment by yosifkit** [comment]
<details>
<summary>Diff:</summary>

```diff
diff --git a/_bashbrew-list b/_bashbrew-list
index 1000895..f612e04 100644
--- a/_bashbrew-list
+++ b/_bashbrew-list
@@ -4,18 +4,18 @@ rust:1-slim
 rust:1-slim-jessie
 rust:1-slim-stretch
 rust:1-stretch
-rust:1.26
-rust:1.26-jessie
-rust:1.26-slim
-rust:1.26-slim-jessie
-rust:1.26-slim-stretch
-rust:1.26-stretch
-rust:1.26.2
-rust:1.26.2-jessie
-rust:1.26.2-slim
-rust:1.26.2-slim-jessie
-rust:1.26.2-slim-stretch
-rust:1.26.2-stretch
+rust:1.27
+rust:1.27-jessie
+rust:1.27-slim
+rust:1.27-slim-jessie
+rust:1.27-slim-stretch
+rust:1.27-stretch
+rust:1.27.0
+rust:1.27.0-jessie
+rust:1.27.0-slim
+rust:1.27.0-slim-jessie
+rust:1.27.0-slim-stretch
+rust:1.27.0-stretch
 rust:jessie
 rust:latest
 rust:slim
diff --git a/rust_jessie/Dockerfile b/rust_jessie/Dockerfile
index f224faa..3e5d97c 100644
--- a/rust_jessie/Dockerfile
+++ b/rust_jessie/Dockerfile
@@ -3,7 +3,7 @@ FROM buildpack-deps:jessie
 ENV RUSTUP_HOME=/usr/local/rustup \
     CARGO_HOME=/usr/local/cargo \
     PATH=/usr/local/cargo/bin:$PATH \
-    RUST_VERSION=1.26.2
+    RUST_VERSION=1.27.0
 
 RUN set -eux; \
     \
diff --git a/rust_latest/Dockerfile b/rust_latest/Dockerfile
index e1b8299..cefddb8 100644
--- a/rust_latest/Dockerfile
+++ b/rust_latest/Dockerfile
@@ -3,7 +3,7 @@ FROM buildpack-deps:stretch
 ENV RUSTUP_HOME=/usr/local/rustup \
     CARGO_HOME=/usr/local/cargo \
     PATH=/usr/local/cargo/bin:$PATH \
-    RUST_VERSION=1.26.2
+    RUST_VERSION=1.27.0
 
 RUN set -eux; \
     \
diff --git a/rust_slim-jessie/Dockerfile b/rust_slim-jessie/Dockerfile
index 7e0e5dc..0c1c347 100644
--- a/rust_slim-jessie/Dockerfile
+++ b/rust_slim-jessie/Dockerfile
@@ -3,7 +3,7 @@ FROM debian:jessie-slim
 ENV RUSTUP_HOME=/usr/local/rustup \
     CARGO_HOME=/usr/local/cargo \
     PATH=/usr/local/cargo/bin:$PATH \
-    RUST_VERSION=1.26.2
+    RUST_VERSION=1.27.0
 
 RUN set -eux; \
     apt-get update; \
diff --git a/rust_slim/Dockerfile b/rust_slim/Dockerfile
index 1bd6725..390dbd9 100644
--- a/rust_slim/Dockerfile
+++ b/rust_slim/Dockerfile
@@ -3,7 +3,7 @@ FROM debian:stretch-slim
 ENV RUSTUP_HOME=/usr/local/rustup \
     CARGO_HOME=/usr/local/cargo \
     PATH=/usr/local/cargo/bin:$PATH \
-    RUST_VERSION=1.26.2
+    RUST_VERSION=1.27.0
 
 RUN set -eux; \
     apt-get update; \
```

</details>

**Comment by yosifkit** [comment]
Build test of #4484; 9cae288d; `amd64` (`rust`):

```console

$ bashbrew build rust:1.27.0-stretch
Building bashbrew/cache:f78c3d026e9580c318028fa164f5c04ca02ed267bddf16871efbae3318a67b1e (rust:1.27.0-stretch)
Tagging rust:1.27.0-stretch
Tagging rust:1-stretch
Tagging rust:1.27-stretch
Tagging rust:stretch
Tagging rust:1.27.0
Tagging rust:1
Tagging rust:1.27
Tagging rust:latest

$ test/run.sh rust:1.27.0-stretch
testing rust:1.27.0-stretch
	'utc' [1/5]...passed
	'cve-2014--shellshock' [2/5]...passed
	'no-hard-coded-passwords' [3/5]...passed
	'override-cmd' [4/5]...passed
	'rust-hello-world' [5/5]...passed


$ bashbrew build rust:1.27.0-slim-stretch
Building bashbrew/cache:3a578f13f9e89d016edd7658bdee1c2aaae6c9631849721d34dbd6f837015d61 (rust:1.27.0-slim-stretch)
Tagging rust:1.27.0-slim-stretch
Tagging rust:1-slim-stretch
Tagging rust:1.27-slim-stretch
Tagging rust:slim-stretch
Tagging rust:1.27.0-slim
Tagging rust:1-slim
Tagging rust:1.27-slim
Tagging rust:slim

$ test/run.sh rust:1.27.0-slim-stretch
testing rust:1.27.0-slim-stretch
	'utc' [1/5]...passed
	'cve-2014--shellshock' [2/5]...passed
	'no-hard-coded-passwords' [3/5]...passed
	'override-cmd' [4/5]...passed
	'rust-hello-world' [5/5]...passed


$ bashbrew build rust:1.27.0-jessie
Building bashbrew/cache:c4f5b964504f6cbba4c0fb5daa598a6ec5116efe7e6194803857db9c2c75d80f (rust:1.27.0-jessie)
Tagging rust:1.27.0-jessie
Tagging rust:1-jessie
Tagging rust:1.27-jessie
Tagging rust:jessie

$ test/run.sh rust:1.27.0-jessie
testing rust:1.27.0-jessie
	'utc' [1/5]...passed
	'cve-2014--shellshock' [2/5]...passed
	'no-hard-coded-passwords' [3/5]...passed
	'override-cmd' [4/5]...passed
	'rust-hello-world' [5/5]...passed


$ bashbrew build rust:1.27.0-slim-jessie
Building bashbrew/cache:7368d4878425c1cd2b25f230145309d4415b93bec0b5bf35aa8c2659ad0cc9c0 (rust:1.27.0-slim-jessie)
Tagging rust:1.27.0-slim-jessie
Tagging rust:1-slim-jessie
Tagging rust:1.27-slim-jessie
Tagging rust:slim-jessie

$ test/run.sh rust:1.27.0-slim-jessie
testing rust:1.27.0-slim-jessie
	'utc' [1/5]...passed
	'cve-2014--shellshock' [2/5]...passed
	'no-hard-coded-passwords' [3/5]...passed
	'override-cmd' [4/5]...passed
	'rust-hello-world' [5/5]...passed

```

**Pull request merged**